Gems/Prefab/PrefabBuilder/PrefabGroup/DefaultProceduralPrefab.cpp

```diff
@@ -0,0 +1,553 @@
+                {
+                    // Check if this transform node is not associated with any meshes
+                    auto parentNodeIndex = graph.GetNodeParent(currentIndex);
+                    const auto parentContent = graph.GetNodeContent(parentNodeIndex);
+                    if (!azrtti_istypeof<AZ::SceneAPI::DataTypes::IMeshData>(parentContent.get()))
+                    {
+                        NodeDataForEntity nodeDataForEntity;
+                        nodeDataForEntity.m_transformIndex = currentIndex;
+                        nodeDataMap.emplace(NodeDataMapEntry{ currentIndex, AZStd::move(nodeDataForEntity) });
+                    }
+                }
+            }
+        }
+
+        return nodeDataMap;
+    }
+
+    bool DefaultProceduralPrefabGroup::AddEditorMaterialComponent(
+        const AZ::EntityId& entityId,
+        const DataTypes::ICustomPropertyData& propertyData) const
+    {
+        const auto propertyMaterialPathIterator = propertyData.GetPropertyMap().find("o3de_default_material");
+        if (propertyMaterialPathIterator == propertyData.GetPropertyMap().end())
+        {
+            // skip these assignment since the default material override was not provided
+            return true;
+        }
+
+        const AZStd::any& propertyMaterialPath = propertyMaterialPathIterator->second;
+        if (propertyMaterialPath.empty() || propertyMaterialPath.is<AZStd::string>() == false)
+        {
+            AZ_Error("prefab", false, "The 'o3de_default_material' custom property value type must be a string."
+                "This will need to be fixed in the DCC tool and re-export the file asset.");
+            return false;
+        }
+
+        // find asset path via node data
+        const AZStd::string* materialAssetPath = AZStd::any_cast<AZStd::string>(&propertyMaterialPath);
+        if (materialAssetPath->empty())
+        {
+            AZ_Error("prefab", false, "Material asset path must not be empty.");
+            return false;
+        }
+
+        // create a material component for this entity's mesh to render with
+        AzFramework::BehaviorComponentId editorMaterialComponent;
+        AzToolsFramework::EntityUtilityBus::BroadcastResult(
+            editorMaterialComponent,
+            &AzToolsFramework::EntityUtilityBus::Events::GetOrAddComponentByTypeName,
+            entityId,
+            "EditorMaterialComponent");
+
+        if (editorMaterialComponent.IsValid() == false)
+        {
+            AZ_Warning("prefab", false, "Could not add the EditorMaterialComponent component; project needs Atom enabled.");
+            return {};
+        }
+
+        // the material product asset such as 'myassets/path/to/cool.azmaterial' is assigned via hint
+        auto materialAssetJson = AZStd::string::format(
+            R"JSON(
+                    {"Controller":{"Configuration":{"materials":[{"Value":{"MaterialAsset":{"assetHint":"%s"}}}]}}}
```

Comment:
Nit: Might be useful to put this in a "pretty" format in case it needs to be updated later